Services/SwissBitcoinPayService.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Net;
+using System.Net.Http;
+using System.Security.Cryptography;
+using System.Text;
+using Newtonsoft.Json;
+using SmartStore.SwissBitcoinPay.Models;
+using SmartStore.SwissBitcoinPay.Settings;
+
+namespace SmartStore.SwissBitcoinPay.Services
+{
+    public class SwissBitcoinPayService
+    {
+
+        public static bool CheckSecretKey(string key, string message, string signature)
+        {
+            var msgBytes = HMACSHA256.HashData(Encoding.UTF8.GetBytes(key), Encoding.UTF8.GetBytes(message));
+            string hashString = string.Empty;
+            foreach (byte x in msgBytes)
+            {
+                hashString += String.Format("{0:x2}", x);
+            }
+            return (hashString == signature);
+        }
+
+        public string CreateInvoice(SwissBitcoinPaySettings settings, PaymentDataModel paymentData)
+        {
+
+            try
+            {
+                var invoice = new SwissBitcoinPayInvoiceModel()
+				{
+                    title = paymentData.Description,
+                    description = $"{paymentData.BuyerName}#Order:{paymentData.OrderID}#Store:{paymentData.StoreID.ToString()}",

```

Comment:
BuyerName seems to be empty, when creating an invoice.
You can also remove the StoreID

```suggestion
                    description = $"{paymentData.BuyerName} | Order {paymentData.OrderID}",

```

Reply:
No, SmartStore Store ID is mandatory by SmartStore.
And the buyer name comes from SmartStore. If it's empty, check the customer settings in SmartStore

Reply:
Ok, at least you can use the | and put spaces around like the suggestion above, to make it more readable.